src/app/MessageDef/TimedRequest.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "TimedRequest.h"
+#include "MessageDefHelper.h"
+
+namespace chip {
+namespace app {
+CHIP_ERROR TimedRequest::Parser::Init(const chip::TLV::TLVReader & aReader)
+{
+    // make a copy of the reader here
+    mReader.Init(aReader);
+    VerifyOrReturnLogError(chip::TLV::kTLVType_Structure == mReader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
+    ReturnLogErrorOnFailure(mReader.EnterContainer(mOuterContainerType));
+    return CHIP_NO_ERROR;
+}
+
+#if CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK
+CHIP_ERROR TimedRequest::Parser::CheckSchemaValidity() const
+{
+    CHIP_ERROR err           = CHIP_NO_ERROR;
+    uint16_t TagPresenceMask = 0;
+    chip::TLV::TLVReader reader;
+    PRETTY_PRINT("TimedRequest =");
+    PRETTY_PRINT("{");
+
+    // make a copy of the reader
+    reader.Init(mReader);
+
+    while (CHIP_NO_ERROR == (err = reader.Next()))
+    {
+        VerifyOrReturnLogError(chip::TLV::IsContextTag(reader.GetTag()), CHIP_ERROR_INVALID_TLV_TAG);
+        switch (chip::TLV::TagNumFromTag(reader.GetTag()))
+        {
+        case kCsTag_Timeout:
+            VerifyOrReturnLogError(!(TagPresenceMask & (1 << kCsTag_Timeout)), CHIP_ERROR_INVALID_TLV_TAG);
+            TagPresenceMask |= (1 << kCsTag_Timeout);
+            VerifyOrReturnLogError(chip::TLV::kTLVType_UnsignedInteger == reader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
+#if CHIP_DETAIL_LOGGING
+            {
+                uint8_t timeout;
+                ReturnLogErrorOnFailure(reader.Get(timeout));
+                PRETTY_PRINT("\tTimeout = 0x%" PRIx8 ",", timeout);
+            }
+#endif // CHIP_DETAIL_LOGGING
+            break;
+        default:
+            ReturnLogErrorOnFailure(CHIP_ERROR_INVALID_TLV_TAG);
+        }
+    }
+    PRETTY_PRINT("}");
+    PRETTY_PRINT("");
+
+    if (CHIP_END_OF_TLV == err)
+    {
+        const uint16_t RequiredFields = (1 << kCsTag_Timeout);
+
+        if ((TagPresenceMask & RequiredFields) == RequiredFields)
+        {
+            err = CHIP_NO_ERROR;
+        }
+    }
+    ReturnLogErrorOnFailure(err);
+    return reader.ExitContainer(mOuterContainerType);
+}
+#endif // CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK
+
+CHIP_ERROR TimedRequest::Parser::GetTimeout(uint8_t * const apTimeout) const
```

Comment:
I don't understand this part.  Per spec timeout is a `uint16`.  See "8.7.4.1. Timed Request Action Information" in the spec.

Yes, the encoding spec has it as 8-bit.  As I keep saying, you should pretty much ignore the encoding spec; it's months behind the IM spec.

Reply:
Updated, file the spec ticket in https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3566

Reply:
Thank you.  Please tag spec issues you file with "SDK Feedback" and the right label for the spec section (in this case "encoding")?